From ee8ac62e815bd430cc5a4661d85d5ff3deaf8593 Mon Sep 17 00:00:00 2001
From: hololeap <hololeap@users.noreply.github.com>
Date: Sun, 8 Jan 2023 12:41:57 -0700
Subject: [PATCH] Add executable flag

The new flag toggles if the htfpp executable is built

Signed-off-by: hololeap <hololeap@users.noreply.github.com>
---
 HTF.cabal | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/HTF.cabal b/HTF.cabal
index c80ac8e..85fb880 100644
--- a/HTF.cabal
+++ b/HTF.cabal
@@ -105,7 +105,14 @@ Custom-Setup
                    process,
                    Cabal
 
+flag executable
+  description:     Build the htfpp executable
+  default:         False
+  manual:          True
+
 Executable htfpp
+  if !flag(executable)
+    buildable: False
   Main-Is:          HTFPP.hs
   Build-Depends:    HUnit,
                     array,
-- 
2.38.2

